pbj-integration-tests/src/test/java/com/hedera/pbj/intergration/test/SampleFuzzTest.java

```diff
@@ -63,18 +103,76 @@ private static Stream<?> objectTestCases() {
                 .flatMap(List::stream);
     }
 
-    @ParameterizedTest
-    @MethodSource("objectTestCases")
-    @Tag(SampleFuzzTest.FUZZ_TEST)
-    void testMethod(Object object) throws Exception {
+    private static record ResultStats(
+            double passRate,
+            double deserializationFailedMean
+    ) {
+        private static final NumberFormat PERCENTAGE_FORMAT = NumberFormat.getPercentInstance();
+
+        boolean passed() {
+            return passRate > PASS_RATE_THRESHOLD
+                    && deserializationFailedMean > DESERIALIZATION_FAILED_MEAN_THRESHOLD;
+        }
+
+        String format() {
+            return "Fuzz tests " + (passed() ? "PASSED" : "FAILED")
+                    + " with passRate = " + PERCENTAGE_FORMAT.format(passRate)
+                    + " and deserializationFailedMean = " + PERCENTAGE_FORMAT.format(deserializationFailedMean);
+        }
+    }
+
+    @Test
+    @Tag(SampleFuzzTest.FUZZ_TEST_TAG)
+    void fuzzTest() {
         assumeFalse(
                 this.getClass().desiredAssertionStatus(),
                 "Fuzz tests run with assertions disabled only. Use the fuzzTest Gradle target."
         );
 
-        FuzzTestResult<?> fuzzTestResult = FuzzTest.fuzzTest(object, THRESHOLD);
-        String resultDescription = fuzzTestResult.format();
-        System.out.println(resultDescription);
-        assertTrue(fuzzTestResult.passed(), resultDescription);
+        final Random random = buildRandom();
+
+        final List<? extends FuzzTestResult<?>> results = objectTestCases()
+                // Note that we must run this stream sequentially to enable
+                // reproducing the tests for a given random seed.
+                .map(object -> FuzzTest.fuzzTest(object, THRESHOLD, random))
+                .peek(result -> System.out.println(result.format()))
```

Comment:
Should it be enabled in debug mode only?

Reply:
Good idea. Just did that in the last comment.